### PR TITLE
Add optional LoL announcer config and polling of live game (#4)

### PR DESCRIPTION
**Pull request by bryangottschalk**
Description: We'd like to start using Riot's live game APIs to build the infrastructure for an alternate LoL announcer that plays custom audio files based on live, in game events.

AC: Based on an optional setting, a polling function hits this endpoint at a set interval of two seconds and logs results to the console or handles the error: http://127.0.0.1:2999/liveclientdata/allgamedata.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -15,11 +15,14 @@ import {
...
 } from './helpers';
+import { pollCurrentGame } from './league-of-legends-api/poll-current-game';
 
 const PORT = process.env.PORT || 8081;
 const app = express();
 
-const DISCORD_BOT_TOKEN = process.env.DISCORD_BOT_TOKEN;
+const DISCORD_BOT_TOKEN: string = process.env.DISCORD_BOT_TOKEN || '';
+const IS_LOL_ANNOUNCER_ENABLED: boolean =
+  Boolean(process.env.LEAGUE_OF_LEGENDS_ANNOUNCER_ENABLED) ?? false;
```
Comment (bryangottschalk) [inline src/index.ts:25]: Thoughts on this being the first iteration for enabling polling, in a future PR refactoring to check if people in voice channels are playing league?

File: src/league-of-legends-api/poll-current-game.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+export const pollCurrentGame = async (
+  channel: VoiceBasedChannel,
+  audioPlayer: AudioPlayer
+) => {
+  setIntervalImmediately(async () => {
+    try {
+      const { data: latestGameResponse }: AxiosResponse<RootGameObject> =
+        await axios.get(`${LOL_GAME_CLIENT_API}/allgamedata`, { httpsAgent });
+
+      const previousKills: number[] =
```
Comment (bryangottschalk) [inline src/league-of-legends-api/poll-current-game.ts:28]: This seems to work well as an initial MVP event, but is pretty hard coded. I'm thinking we can use the Event array from the Riot API moving forward to build something more sophisticated. 

image

File: src/league-of-legends-api/types/index.ts [hunk 3]
```diff
@@ -0,0 +1,246 @@
+export interface RootGameObject {
+  activePlayer: ActivePlayer;
+  allPlayers: Player[];
+  events: Events;
+  gameData: GameData;
+}
+
+export interface E {
```
Comment (bryangottschalk) [inline src/league-of-legends-api/types/index.ts:8]: FYI I didnt write this by hand, I dropped the JSON into http://json2ts.com/

File: src/index.ts [hunk 4]
```diff
@@ -48,8 +51,11 @@ const audioPlayer = createAudioPlayer();
 
 // Event triggered when the audio player goes idle (i.e. not playing anything)
 audioPlayer.on(AudioPlayerStatus.Idle, () => {
+  // Restore this logic to disconnect bot after it announces clips. Removing it causes the bot to stay in the channel
```
Comment (bryangottschalk) [inline src/index.ts:54]: This is a notable change

File: src/index.ts (hunk 1 above)
Comment (walworob) [reply src/index.ts:25]: Yeah I think this is solid for now. I captured a couple thoughts on what we could do long-term with this in #6 .

File: src/index.ts (hunk 4 above)
Comment (walworob) [reply src/index.ts:54]: With the commands now merged we can use `!gtfo` and `!cmere` to get the bot to move between voice channels if needed as well. Also this puts #3 on hold until we figure out exactly how we want to handle bot disconnects.

File: src/league-of-legends-api/poll-current-game.ts (hunk 2 above)
Comment (walworob) [reply src/league-of-legends-api/poll-current-game.ts:28]: Yeah this is where we'd get the multi-kill data anyways so makes sense to use that. I think looking at events we'd only have to keep track of the number of events in the previous game data to determine if there are new ones and then parse those.

**Review by walworob — APPROVED** [review]
LGTM, lots we can work with here moving forward!

File: src/index.ts (hunk 1 above)
Comment (bryangottschalk) [reply src/index.ts:28]: Awesome thanks for doing that

**Pull request merged**